File: examples/university-demo/src/main/java/org/axonframework/examples/demo/university/faculty/write/subscribestudentmulti/SubscribeStudentToCourse.java
package org.axonframework.examples.demo.university.faculty.write.subscribestudentmulti;

import org.axonframework.examples.demo.university.shared.ids.CourseId;
import org.axonframework.examples.demo.university.shared.ids.StudentId;

public record SubscribeStudentToCourse(StudentId studentId, CourseId courseId) {

}

File: examples/university-demo-kotlin/src/main/kotlin/faculty/write/subscribe_student_polymorph/SubscribeStudentToCourse.kt
package org.axonframework.examples.university.faculty.write.subscribe_student_polymorph

import org.axonframework.examples.university.shared.ids.CourseId
import org.axonframework.examples.university.shared.ids.StudentId
import org.axonframework.examples.university.shared.ids.SubscriptionId
import org.axonframework.messaging.commandhandling.annotation.Command
import org.axonframework.modelling.annotation.TargetEntityId

@Command
data class SubscribeStudentToCourse(
    val studentId: StudentId,
    val courseId: CourseId,
) {
    @TargetEntityId
    val subscriptionId: SubscriptionId = SubscriptionId(studentId, courseId)
}
